File: myExpenses/src/main/java/org/totschnig/myexpenses/preference/SimplePasswordDialogFragmentCompat.java
package org.totschnig.myexpenses.preference;

import android.os.Bundle;
import android.support.design.widget.TextInputEditText;
import android.support.v7.preference.PreferenceDialogFragmentCompat;
import android.view.View;

import org.totschnig.myexpenses.R;

public class SimplePasswordDialogFragmentCompat extends PreferenceDialogFragmentCompat {

  private TextInputEditText passwordInput;

  public static SimplePasswordDialogFragmentCompat newInstance(String key) {
    SimplePasswordDialogFragmentCompat fragment = new SimplePasswordDialogFragmentCompat();
    Bundle bundle = new Bundle(1);
    bundle.putString(ARG_KEY, key);
    fragment.setArguments(bundle);
    return fragment;
  }

  @Override
  protected void onBindDialogView(View view) {
    super.onBindDialogView(view);
    passwordInput = view.findViewById(R.id.passwordEdit);
    String currentValue = ((SimplePasswordPreference) getPreference()).getValue();
    if (currentValue != null) {
      passwordInput.setText(currentValue);
    }
  }

  @Override
  public void onDialogClosed(boolean positiveResult) {
    final SimplePasswordPreference preference = (SimplePasswordPreference) getPreference();

    if (!positiveResult) {
      return;
    }
    preference.setValue(passwordInput.getText().toString());
  }
}

File: myExpenses/src/main/java/org/totschnig/myexpenses/preference/SimplePasswordDialogFragmentCompat.kt
package org.totschnig.myexpenses.preference

import android.os.Bundle
import android.view.View
import androidx.preference.PreferenceDialogFragmentCompat
import com.google.android.material.textfield.TextInputEditText
import org.totschnig.myexpenses.R

class SimplePasswordDialogFragmentCompat : PreferenceDialogFragmentCompat() {
    private lateinit var passwordInput: TextInputEditText
    override fun onBindDialogView(view: View) {
        super.onBindDialogView(view)
        passwordInput = view.findViewById(R.id.passwordEdit)
        val currentValue = (preference as SimplePasswordPreference).value
        if (currentValue != null) {
            passwordInput.setText(currentValue)
        }
    }

    override fun onDialogClosed(positiveResult: Boolean) {
        val preference = preference as SimplePasswordPreference
        if (!positiveResult) {
            return
        }
        preference.value = passwordInput.text.toString()
    }

    companion object {
        @JvmStatic
        fun newInstance(key: String) = SimplePasswordDialogFragmentCompat().apply {
                arguments = Bundle(1).apply {
                    putString(ARG_KEY, key)
                }
            }
    }
}